### PR TITLE
net/portmapper: handle pcp ADDRESS_MISMATCH response

### PR DESCRIPTION
**Pull request by josharian**
These show up a fair amount in our logs.

Signed-off-by: Josh Bleecher Snyder <josh@tailscale.com>

### CODE REVIEW

File: net/portmapper/portmapper.go [hunk 1]
```diff
@@ -791,6 +791,10 @@ func (c *Client) Probe(ctx context.Context) (res ProbeResult, err error) {
 						// provide port mapping services.
 						res.PCP = false
 						continue
+					case pcpCodeAddressMismatch:
+						// A PCP service is running, but it is behind a NAT, so it can't help us.
```
Comment (bradfitz) [inline net/portmapper/portmapper.go:795]: want to add some clientmetrics for these paths too?

**Review by bradfitz — APPROVED** [review]

File: net/portmapper/portmapper.go (hunk 1 above)
Comment (josharian) [reply net/portmapper/portmapper.go:798]: Done.

**Comment by josharian** [comment]
And also made the result codes stringers.

File: net/portmapper/portmapper.go [hunk 2]
```diff
@@ -835,3 +857,45 @@ var uPnPPacket = []byte("M-SEARCH * HTTP/1.1\r\n" +
 	"ST: ssdp:all\r\n" +
 	"MAN: \"ssdp:discover\"\r\n" +
 	"MX: 2\r\n\r\n")
+
+var (
```
Comment (bradfitz) [inline net/portmapper/portmapper.go:861]: this does PMP and PCP... want to do UPnP while you're here?

also, send a metric for each about whether we send a discovery/probe request at all? would be useful for battery debugging.

**Review by bradfitz — APPROVED** [review]

File: net/portmapper/portmapper.go (hunk 2 above)
Comment (josharian) [reply net/portmapper/portmapper.go:861]: ok. next week, i'm out of typing for today.

Comment (bradfitz) [reply net/portmapper/portmapper.go:861]: Ping, if you forgot this.

Comment (josharian) [reply net/portmapper/portmapper.go:861]: Thanks for the reminder. Now with UPnP metrics, a few more PXP metrics, and a bonus minor fix.

**Review by bradfitz — APPROVED** [review]

**Pull request merged**